### PR TITLE
libguestfs: fix perl commands

### PR DESCRIPTION
**Pull request by ljli**
###### Motivation for this change
Commands depending on perl such as `virt-list-partitions` or `virt-list-filesystems` were not working, as some dependencies couldn't be found by perl.

###### Things done
Provide dependencies via `PERL5LIB` wrapper.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libguestfs/default.nix [hunk 1]
```diff
@@ -18,13 +18,15 @@ stdenv.mkDerivation rec {
...
+  perlDeps = with perlPackages; [ perl libintl_perl GetoptLong ModuleBuild SysVirt ];
   nativeBuildInputs = [ autoreconfHook makeWrapper pkgconfig ];
   buildInputs = [
     ncurses cpio gperf jansson
     cdrkit flex bison qemu pcre augeas libxml2 acl libcap libcap_ng libconfig
     systemd fuse yajl libvirt gmp readline file hivex
-    numactl xen libapparmor getopt perlPackages.ModuleBuild
-  ] ++ (with perlPackages; [ perl libintl_perl GetoptLong SysVirt ])
+    numactl xen libapparmor getopt
+    perlPackages.perl
```
Comment (volth) [inline pkgs/development/libraries/libguestfs/default.nix]: `perl` is already included via `perlDeps`, so this line can be removed

Comment (ljli) [reply pkgs/development/libraries/libguestfs/default.nix]: Removed `perl` from `perlDeps`.

**Comment by ljli** [comment]
I addressed the review comment.
Regarding refactoring, I tried to use `perl.withPackages`. The libguestfs perl commands depend on a perl module path in the derivation itself (second `--prefix PERL5LIB` argument to `makeWrapper`). But `perl.withPackages` produces a wrapped `perl` that simply sets `PERL5LIB`, overwriting the inherited `PERL5LIB` content without adding to it. I wasn't sure how to deal with that.

**Comment by volth** [comment]
> But perl.withPackages produces a wrapped perl that simply sets PERL5LIB, overwriting the inherited PERL5LIB content without adding to it. I wasn't sure how to deal with that.

Oh, it seems like a my bug.

**Comment by volth** [comment]
@ljli (or me, or anyone who might go the refactor way): besides #78118 it will require #59976-like fix for `hivex`

**Comment by JohnAZoidberg** [comment]
Seems `virt-list-partitions` or `virt-list-filesystems` have been replaced: https://libguestfs.org/virt-filesystems.1.html
Is this PR still needed?

**Comment by JohnAZoidberg** [comment]
In case we still need it, I merged master into it, to fix the conflicts.

**Comment by twz123** [comment]
I've opened a version update for libguestfs which addresses the issue with the Perl bindings: #185015. Note that as of [libguestfs 1.46](https://github.com/libguestfs/libguestfs/blob/ab2d624f46b6cb69a332f513ea32a1f7c9170ca5/docs/guestfs-release-notes-1.46.pod), many of the CLI tools (including virt-filesystems) have moved to the new [guestfs-tools](https://github.com/rwmjones/guestfs-tools) project. That one is not yet packaged for NixOS, but I'm giving it a stab right now.

**Pull request closed without merging**